libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLine.java

```diff
@@ -75,17 +77,26 @@
   private int primaryRouteIndex;
   private boolean isVisible = true;
   private boolean alternativesVisible = true;
+  private AtomicReference<FeatureProcessingTask> featureProcessingTaskRef = new AtomicReference<>(null);
+  private FeatureProcessingTask featureProcessingTask;
```

Comment:
I got your intent to use `featureProcessingTask`, but my question is: Do we have option to not use additional variable (and some methods below) just for tests purposes? If no - I'm ok with it) But if we have, I think it would be great to avoid it, because I think we should not increase codebase just due testing goals (where possible of course). WDYT?

Reply:
You're completely right. Testing async code is hard 😂 Previous tests relied in Robolectric to run the `AsyncTask`s and we didn't need to control the asynchronicity ourselves. We do now, and we need to "inject" somehow the tasks so previous tests pass - as we changed from `AsyncThread` to `Thread`. We can revisit this in a future / separate PR.